spec/support/appear_before_matcher.rb

```diff
@@ -0,0 +1,32 @@
+module SortOrderMatchers
```

Comment:
I went looking for it on the internet, and rolled my own based on someone else's code which did 200% of what we needed here. Custom Matchers are the best.